Docker build with `TARGET=container` fails because of incorrect version of `go` in v1.12.4

### Discussed in https://github.com/nginxinc/kubernetes-ingress/discussions/2609

Error:
```
 > [builder 6/6] RUN CGO_ENABLED=0 GO111MODULE=on GOOS=linux GOARCH=amd64 go build -trimpath -ldflags "-s -w -X main.version=v1.12.4-SNAPSHOT-addceb8 -X main.commit=addceb86fb4cfa225696a1f88f649d625e5e8857 -X main.date=2022-04-14T16:22:08Z" -o /nginx-ingress:
#19 26.82 # sigs.k8s.io/json/internal/golang/encoding/json
#19 26.82 /go/pkg/mod/sigs.k8s.io/json@v0.0.0-20211020170558-c049b76a60c6/internal/golang/encoding/json/encode.go:1249:12: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
#19 26.82 /go/pkg/mod/sigs.k8s.io/json@v0.0.0-20211020170558-c049b76a60c6/internal/golang/encoding/json/encode.go:1255:18: sf.IsExported undefined (type reflect.StructField has no field or method IsExported)
```

The Docker build fails when using `TARGET=container`. This happens because the build container uses `go` 1.16 and the project was updated to use `go` 1.17

--- a/build/Dockerfile
+++ b/build/Dockerfile
@@ -247,7 +247,7 @@ LABEL org.nginx.kic.image.build.nginx.version="${NGINX_PLUS_VERSION}${NGINX_VERS
 
 
 ############################################# Build nginx-ingress in golang container #############################################
-FROM golang:1.16-alpine AS builder
+FROM golang:1.17-alpine AS builder
 ARG IC_VERSION
 ARG GIT_COMMIT
 ARG DATE

--- a/tests/test-servers/Dockerfile
+++ b/tests/test-servers/Dockerfile
@@ -1,4 +1,4 @@
-FROM golang:1.16-alpine as builder
+FROM golang:1.17-alpine as builder
 
 RUN mkdir /app
 

